spec/integration/proxy/api_resolver_spec.lua

```diff
@@ -194,6 +198,26 @@ describe("Resolver", function()
         local upstream_url = body.log.entries[1].request.url
         assert.equal("http://mockbin.com/har/of/request", upstream_url)
       end)
+      it("should not add a trailing slash when strip_path is enabled", function()
+        local response, status = http_client.get(spec_helper.PROXY_URL.."/test-trailing-slash", { hello = "world"})
+        assert.equal(200, status)
+        assert.equal("http://www.mockbin.org/request?hello=world", cjson.decode(response).url)
+      end)
+      it("should not add a trailing slash when strip_path is disabled", function()
+        local response, status = http_client.get(spec_helper.PROXY_URL.."/test-trailing-slash2", { hello = "world"})
+        assert.equal(200, status)
+        assert.equal("http://www.mockbin.org/request/test-trailing-slash2?hello=world", cjson.decode(response).url)
+      end)
+      it("should not add a trailing slash when strip_path is enabled and upstream_url has no path", function()
+        local response, status = http_client.get(spec_helper.PROXY_URL.."/test-trailing-slash3/request", { hello = "world"})
+        assert.equal(200, status)
+        assert.equal("http://www.mockbin.org/request?hello=world", cjson.decode(response).url)
+      end)
+      it("should not add a trailing slash when strip_path is enabled and upstream_url has single path", function()
+        local response, status = http_client.get(spec_helper.PROXY_URL.."/test-trailing-slash4/request", { hello = "world"})
+        assert.equal(200, status)
+        assert.equal("http://www.mockbin.org/request?hello=world", cjson.decode(response).url)
+      end)
```

Comment:
Would be nice if some more use cases could be described in the unit tests too: https://github.com/Mashape/kong/blob/next/spec/unit/core/resolver_spec.lua